### DISCUSSION TITLE
当有两个域名反向代理时，site_url如何设定？

### DISCUSSION THREAD

**Discussion by liaoyungchun** [AlistGo/alist#6927] [Q&A] [answered]
有两个域名需要反向代理到同一个alist，比如  x.domain.com和 x4.domain.com（后者用于ipv4访问）

配置中的site_url设为 https://x.domain.com，

访问 https://x4.domain.com时：

在文件列表上右击文件下载，下载链接地址是 https://x4.domain.com/....
当点击此文件进去，再点击“下载按钮“，下载链接地址却是： https://x.domain.com/....

二者不一样，会导致无法下载，或下载慢的问题，也令人迷惑。希望能匹配访问地址。

**Comment by liaoyungchun** [comment]
补充一下，"复制链接按钮"与右键"下载"的链接是一致的，只有"下载按钮"的链接是不正确的

**Comment by Simonzhang8** [comment]
要不就不填site_url试试？

**Reply by liaoyungchun** [reply]
试过了，不填时前面是 http://[::]:5244，下载404，所以必须填。

这个实际是个bug，是下载按钮链接错误，跟其它两处的链接不一致，应该比较好修复的。

**Reply by Simonzhang8** [reply]
> 试过了，不填时前面是 http://[::]:5244，下载404，所以必须填。
> 
> 这个实际是个bug，是下载按钮链接错误，跟其它两处的链接不一致，应该比较好修复的。

我这里就没有出现这个问题啊，两个域名都有各自的下载地址，ip地址也有ip地址的下载地址
想问下你反向代理是怎么配置的？

**Reply by liaoyungchun** [reply]
> > 试过了，不填时前面是 http://[::]:5244，下载404，所以必须填。
> > 这个实际是个bug，是下载按钮链接错误，跟其它两处的链接不一致，应该比较好修复的。
> 
> 我这里就没有出现这个问题啊，两个域名都有各自的下载地址，ip地址也有ip地址的下载地址 想问下你反向代理是怎么配置的？

我nginx是这么配的：

    location / {
                proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
                proxy_set_header X-Forwarded-Proto $scheme;
                proxy_set_header Host $http_host;
                proxy_set_header X-Real-IP $remote_addr;
                proxy_set_header Range $http_range;
                proxy_set_header If-Range $http_if_range;
                proxy_redirect off;
                proxy_pass http://[::]:5244/; #这是我的配置端口
                #the max size of file to upload
                client_max_body_size 200G;
    }

**Reply by Simonzhang8** [reply, marked as answer]
已经成功复现你的情况了，问题就是反代时没有传递正确的host
参考：#726 如果你用的是nginx proxy manager的话需要加上这一句，否则host传递不生效
``` 
proxy_set_header X-Forwarded-Host $http_host; 
```

**Reply by liaoyungchun** [reply]
> 已经成功复现你的情况了，问题就是反代时没有传递正确的host 参考：#726 如果你用的是nginx proxy manager的话需要加上这一句，否则host传递不生效
> 
> ```
> proxy_set_header X-Forwarded-Host $http_host; 
> ```

非常感谢，这个是正解，且 site_url 不设置，现在正常了！
